models/db.go

```diff
@@ -34,42 +29,56 @@ func openConnection(service, dsn string) (gorm.Dialector, error) {
 	return open(dsn), nil
 }
 
-func (d *db) Connection() *gorm.DB {
-	if conn == nil {
-		dialect, err := openConnection(settings.DatabaseService, settings.DatabaseDSN)
+func (d *DB) Connection() *gorm.DB {
+	config := d.Config
+	if d.conn == nil {
```

Comment:
OPTIONAL: don't you think there is an extra nesting that can be avoided with an early return?

```
if d.conn != nil {
  return d.conn
}

// DB setup code goes here
```
https://go.dev/wiki/CodeReviewComments#indent-error-flow
https://youtu.be/CFRhGnuXG-4?si=3voZKjZ_4v8AePYh